pom.xml

```diff
@@ -167,9 +206,13 @@
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>org.apache.httpcomponents</groupId>
-      <artifactId>httpclient</artifactId>
-      <version>4.5.14</version>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>gson-api</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>apache-httpcomponents-client-4-api</artifactId>
       <scope>test</scope>
```

Comment:
why are these added?
this would incorrectly cause these plugins to be added into the Jenkins used by `RealJenkinsRule`, which would be unwanted and could potentially mask some issues.
Noting we do not have any `RealJenkinsRule`s currently - but this is at least very strange to me, and has the potential to increase faileure in the PCT in the future.

Reply:
Elsewhere I have had problems with upper bounds between (a) a library coming in through a Jenkins plugin that is a transitive dependency of a direct Jenkins plugin dependency and (b) a test-scoped direct library dependency. Using the Jenkins plugin consistently in these cases avoids the upper bounds errors.

However, I do not feel strongly about this one way or another. In the interest of time, I am removing this change to avoid another round-trip code review.

Reply:
Resolved in commit 82d7e6f, as the relevant code is no longer present in the PR.

Reply:
thankyou for the explanation